Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,258 @@
+                {
+                    SetNames(ref name, ref maxBalance, peopleBalances, balance);
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word1 = name.ToString().Contains(", ") ? "are" : "is";
+            var word2 = name.ToString().Contains(", ") ? "people" : "person";
+            return $"{FixPlural(name.ToString())} {word1} the richest {word2}. {maxBalanceNoComma}.";
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public string FindMostPoorPerson(string[] peopleAndBalances)
+        {
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return NotAvailable;
+            }
+
+            var minBalance = decimal.MaxValue;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid)
+                {
+                    GetPoorest(ref name, ref minBalance, peopleBalances, balance);
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var minBalanceNoComma = RemoveComma($"{minBalance:C0}").Replace("(", "-").Replace(")", "");
+            var word = name.ToString().Contains(", ") ? "have" : "has";
+            return $"{FixPlural(name.ToString())} {word} the least money. {minBalanceNoComma}.";
+        }
+
+        private void GetPoorest(ref StringBuilder mostPoorPerson, ref decimal minBalance, string[] peopleBalances, decimal balance)
+        {
+            if (balance <= minBalance)
+            {
+                if (balance < minBalance)
+                {
+                    mostPoorPerson = new StringBuilder();
+                }
+
+                if (mostPoorPerson.Length != 0)
+                {
+                    mostPoorPerson.Append(", ");
+                }
+
+                mostPoorPerson.Append(peopleBalances[0]);
+                minBalance = balance;
+            }
+        }
+
+        private string FixPlural(string name)
+        {
+            var lastComma = name.LastIndexOf(", ", StringComparison.InvariantCulture);
+
+            if (lastComma > 0)
+            {
+                name = name.Remove(lastComma, 2).Insert(lastComma, " and ");
+            }
+
+            return name;
+        }
+
+        private bool IsArrayNullOrEmpty(string[] array)
+        {
+            if (array == null || array.Length == 0)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private string RemoveComma(string data)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'RemoveComma' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)